onnx/defs/nn/defs.cc

```diff
@@ -23,6 +23,94 @@ static std::string auto_pad_doc =
 } // namespace ONNX_NAMESPACE
 
 namespace ONNX_NAMESPACE {
+
+void convPoolTypeAndShapeInference(InferenceContext& ctx) {
+  propagateElemTypeFromInputToOutput(ctx, 0, 0);
+
+  if (!hasNInputShapes(ctx, 2)) {
+    return;
+  }
+
+  // don't bother with legacy auto_pad for now
+  if (ctx.getAttribute("auto_pad")) {
+    return;
+  }
+
+  size_t n_input_dims = (size_t) (ctx.getInputType(0)->tensor_type().shape().dim_size() - 2);
+
+  std::vector<int64_t> dilations;
+  if (getRepeatedAttribute(ctx, "dilations", dilations)) {
+    if (dilations.size() != n_input_dims) {
+      return;
+    }
+  } else {
+    for (size_t i = 0; i < n_input_dims; ++i) {
+      dilations.push_back(1);
+    }
+  }
+
+  std::vector<int64_t> kernel_shape;
+  if (getRepeatedAttribute(ctx, "kernel_shape", kernel_shape)) {
+    if (kernel_shape.size() != ctx.getInputType(1)->tensor_type().shape().dim_size() - 2) {
```

Comment:
getInputType(0)?

Reply:
yeah, that will work. I guess there's a little more difference between Conv and Pool than I was thinking